lib/init/testsuite/test_grass_tmp_mapset.py

```diff
@@ -30,13 +30,16 @@ class TestTmpMapset(unittest.TestCase):
     """Tests --tmp-mapset option of grass command"""
 
     # TODO: here we need a name of or path to the main GRASS GIS executable
-    executable = "grass"
+    executable = shutil.which("grass")
```

Comment:
I would think only shutil.which or shell=True are needed for Windows, not both. I would prefer shutil.which or some equivalent of sys.executable in the _grass_ package (e.g., grass.script.setup.executable).

Reply:
I see. Unfortunately, both shutil.which and shell do not seem to help. Maybe better to switch to subprocess.run?

Reply:
subprocess.run is just an interface to Popen like all the others. I don't expect it would help. shutil.which needs `grass` to be on path to work which is likely not fulfilled.

This test is now unique - we don't have other tests which test the executable - we we may have more in the future or even write such code, so perhaps ignoring it now and adding `grass.script.setup.executable` later is the way to go.

Reply:
This is one the of things for a separate PR. Getting the _grass.script.setup.executable_ piece working would be nice to have. [get_install_path](https://github.com/OSGeo/grass/blob/main/python/grass/script/setup.py#L115) is perhaps a good starting point (Python package path is known, path to installation is not). Lazy-loading seems appropriate, but there are still no properties for modules yet (?), but there is [__getattr__](https://peps.python.org/pep-0562/).